packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -371,13 +355,11 @@ export class ProjectActions {
         : codeGenCandidate
     }
 
-    const specFileExtension = getFileExtension()
     const codeGenPath = getCodeGenPath()
 
     const newSpecCodeGenOptions = new SpecOptions(this.ctx, {
       codeGenPath,
       codeGenType,
-      specFileExtension,
```

Comment:
I moved the specFileExtension logic into SpecOptions to:
1. co-locate it with the relevant logic
2. make it a little more testable